services/121-service/src/registration/services/bulk-import.service.ts

```diff
@@ -380,4 +389,100 @@ export class BulkImportService {
     }
     return validatatedArray;
   }
+
+  public async importFspReconciliation(
+    csvFile,
+    programId: number,
+    userId: number,
+  ): Promise<ImportResult> {
+    const validatedImportRecords = await this.csvToValidatedFspReconciliation(
+      csvFile,
+    );
+
+    let countImported = 0;
+    let countNotFound = 0;
+
+    const importResponseRecords = [];
+    for await (const record of validatedImportRecords) {
+      const importResponseRecord = record as UploadFspReconciliationResult;
+
+      const registration = await this.registrationRepository.findOne({
+        where: { referenceId: record.referenceId },
+        relations: ['fsp'],
+      });
+      if (!registration) {
+        importResponseRecord.importStatus = ImportStatus.unmatched;
+        importResponseRecords.push(importResponseRecord);
+        countNotFound += 1;
+        continue;
+      }
+
+      const paTransactionResult = new PaTransactionResultDto();
```

Comment:
What happens if you import the same file twice? You get multiple of the same transactions with later timestamps in the database?

Reply:
Yes, but that's no problem right? That's how our transactions system is set up, you can send as many status updates as you want, and always the latest one is taken. Right?

Reply:
But then the transaction date changes right. Do we not use that somewhere? Don't we want to differentiate between PAs that received their payment during an earlier import?

Reply:
Sure. I've made task to not update transaction if status stays the same.